### ISSUE TITLE
Hauskeeping test - FileAlreadyExist

### ISSUE THREAD

**Issue by zdenek-jonas** [microstream-one/microstream#141]
The Housekeeping test writes an error in the console and then freezes in an never ending loop..

project:
Microstream-test
branch: master
Test location: microstream-test/intern/src/main/test/java/test/microstream/HouseKeepingIntervalTest.java

error stack:
```
Exception in thread "MicroStream-StorageChannel-0" one.microstream.storage.exceptions.StorageExceptionIo
	at one.microstream.storage.types.StorageFileWriter.rescueFromDeletion(StorageFileWriter.java:231)
	at one.microstream.storage.types.StorageFileWriter.deleteFile(StorageFileWriter.java:171)
	at one.microstream.storage.types.StorageFileWriter.delete(StorageFileWriter.java:161)
	at one.microstream.storage.types.StorageFileManager$Default.deleteFile(StorageFileManager.java:1543)
	at one.microstream.storage.types.StorageLiveDataFile$Default.lambda$1(StorageLiveDataFile.java:242)
	at one.microstream.storage.types.StorageLiveFile$Abstract.unregisterUsageClosing(StorageLiveFile.java:201)
	at one.microstream.storage.types.StorageLiveDataFile$Default.unregisterUsageClosingData(StorageLiveDataFile.java:241)
	at one.microstream.storage.types.StorageFileManager$Default.incrementalDissolveStorageFile(StorageFileManager.java:1503)
	at one.microstream.storage.types.StorageFileManager$Default.internalCheckForCleanup(StorageFileManager.java:1448)
	at one.microstream.storage.types.StorageFileManager$Default.incrementalFileCleanupCheck(StorageFileManager.java:1356)
	at one.microstream.storage.types.StorageChannel$Default.houseKeepingCheckFileCleanup(StorageChannel.java:243)
	at one.microstream.storage.types.StorageChannel$Default.houseKeeping(StorageChannel.java:216)
	at one.microstream.storage.types.StorageChannel$Default.work(StorageChannel.java:301)
	at one.microstream.storage.types.StorageChannel$Default.run(StorageChannel.java:353)
	at java.lang.Thread.run(Thread.java:748)
Caused by: one.microstream.exceptions.IORuntimeException: java.nio.file.FileAlreadyExistsException: tmp/junit6767749768429913569/deleted/channel_0/channel_0_1_2020-07-27_15-28-51_151.bak
	at one.microstream.afs.nio.NioIoHandler$Default.specificTargetMoveFile(NioIoHandler.java:693)
	at one.microstream.afs.nio.NioIoHandler$Default.specificMoveFile(NioIoHandler.java:668)
	at one.microstream.afs.nio.NioIoHandler$Default.specificMoveFile(NioIoHandler.java:1)
	at one.microstream.afs.AIoHandler$Abstract.moveFile(AIoHandler.java:991)
	at one.microstream.afs.AWritableFile.moveTo(AWritableFile.java:100)
	at one.microstream.storage.types.StorageFile$Abstract.moveTo(StorageFile.java:377)
	at one.microstream.storage.types.StorageFileWriter.lambda$1(StorageFileWriter.java:225)
	at one.microstream.afs.AFS.executeWriting(AFS.java:255)
	at one.microstream.afs.AFS.executeWriting(AFS.java:242)
	at one.microstream.storage.types.StorageFileWriter.rescueFromDeletion(StorageFileWriter.java:222)
	... 14 more
Caused by: java.nio.file.FileAlreadyExistsException: tmp/junit6767749768429913569/deleted/channel_0/channel_0_1_2020-07-27_15-28-51_151.bak
	at sun.nio.fs.UnixCopyFile.move(UnixCopyFile.java:429)
	at sun.nio.fs.UnixFileSystemProvider.move(UnixFileSystemProvider.java:262)
	at java.nio.file.Files.move(Files.java:1395)
	at one.microstream.io.XIO.move(XIO.java:903)
	at one.microstream.afs.nio.NioIoHandler$Default.specificTargetMoveFile(NioIoHandler.java:686)
	... 23 more

```
test:
```
class HouseKeepingIntervalTest {

    @TempDir
    Path location;

    @Test
    void houseKeepingIntervalTest() throws InterruptedException {
        Customer customer = CustomerGenerator.generateNewCustomer();

        Configuration configuration = Configuration.Default();

        EmbeddedStorageManager storageManager = configuration.setBaseDirectory(location.toString())
                .setDataFileMaximumSize(2048)
                .setDataFileMinimumSize(1024)
                .setDeletionDirectory(location.resolve("deleted").toString())
                .setHousekeepingIntervalMs(10)
                .createEmbeddedStorageFoundation().createEmbeddedStorageManager(customer).start();

        for (int i = 0; i < 100; i++) {
            customer = CustomerGenerator.generateNewCustomer();
            storageManager.store(customer);
        }

        Thread.sleep(1_000);

        List<File> files = (List<File>) FileUtils.listFiles(location.resolve("deleted").toFile(), null, true);
        assertTrue(files.size() > 2, files.toString());

        storageManager.shutdown();

    }
}
```

**Comment by zdenek-jonas** [comment]
DeletionDirectoryTest.java - same problem

**Comment by tm-ms** [comment]
After the usual amount of time for dancing with Maven, I finally got to reproduce the exception.

There simply was the tiny bug that the file moving logic ensures the existence of the target file but the NIO file moving function does not replace existing files implicitely, but throws an exception.
I removed the ensuring call and now the test passes on my machine.

Side note:
There was no loop.
The main thread was waiting to be notified by the storage thread(s) but the storage thread died on the file moving exception.

Granted, it is weird that such a hangup can happen.
I looked into that and found:
- If the currently executed task (store, load, export, etc.) fails, there's a try-catch handling that and preventing the storage channel thread from dying.
- However, the housekeeping work is outside of that and if any housekeeping action causes an exception, it causes the channel thread to terminate, leaving the application waiting for its response into eternity.

The Housekeeping call needs a similar mechanism to protect the thread from dying.
Or it has to trigger a notification for the waiting thread and shutting down the storage completely.
(Because who wants to keep the storage management logic running after something went wrong in it ...)
I'll add an issue and a code task for that.
